qst/src/main/java/io/deephaven/qst/table/DropColumnsTable.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.qst.table;
+
+import io.deephaven.annotations.NodeStyle;
+import io.deephaven.api.ColumnName;
+import org.immutables.value.Value.Check;
+import org.immutables.value.Value.Default;
+import org.immutables.value.Value.Immutable;
+
+import java.util.List;
+
+@Immutable
+@NodeStyle
+public abstract class DropColumnsTable extends TableBase implements SingleParentTable {
+
+    public static Builder builder() {
+        return ImmutableDropColumnsTable.builder();
+    }
+
+    public abstract TableSpec parent();
+
+    public abstract List<ColumnName> dropColumns();
+
+    @Override
+    public final <V extends Visitor> V walk(V visitor) {
+        visitor.visit(this);
+        return visitor;
+    }
+
+    @Check
+    final void checkNonEmpty() {
+        if (dropColumns().isEmpty()) {
+            throw new IllegalArgumentException("Expected dropColumns() to be non-empty");
+        }
+    }
```

Comment:
This feels controversial.

Reply:
Hmm. Would be reasonable to add these 3 lines at the top of `QueryTable.dropColumns`:
```
        if (columnNames.length == 0) {
            return prepareReturnThis();
        }
```